ansible/roles/base/tasks/07grub.yml

```
@@ -0,0 +1,8 @@
+---
+- name: set kernel command line options
+  lineinfile:
+    path: /etc/default/grub
+    regexp: '^#?GRUB_CMDLINE_LINUX='
+    line: 'GRUB_CMDLINE_LINUX="{% for opt,val in (install.kernel_cmdline | default({})).items() %}{{ opt }}={{ val }}{% if not loop.last %} {% endif %}{% endfor %}"'
```

```suggestion
    line: 'GRUB_CMDLINE_LINUX="{{ map(lambda opt, val: opt + '=' + val, (install.kernel_cmdline or {}).items()).join(' ') }}"'
```

I'm not sure I like it all that much more than the explicit loop, but the `if loop.end` stuff is gnarly.

---

yeah you are right...

---

actually i just realized that not all command line arguments need to have a value ... so we should probably change the map to a list....

---

Oh, even simpler then.  Going ahead with that.

---

OK, done.